apiserver/crossmodel/base_test.go

```diff
@@ -30,7 +37,44 @@ func (s *baseCrossmodelSuite) SetUpTest(c *gc.C) {
 	s.resources = common.NewResources()
 	s.authorizer = testing.FakeAuthorizer{names.NewUserTag("testuser"), true}
 
+	s.calls = []string{}
 	var err error
-	s.api, err = crossmodel.NewAPI(nil, s.resources, s.authorizer)
+	s.initialiseExporter()
+	s.api, err = apicrossmodel.CreateAPI(s.exporter, s.resources, s.authorizer)
 	c.Assert(err, jc.ErrorIsNil)
 }
+
+func (s *baseCrossmodelSuite) initialiseExporter() {
+	s.exporter = &mockExporter{}
+	s.exporter.exportOffer = func(offer crossmodel.Offer) error {
+		s.calls = append(s.calls, exportOfferCall)
```

Comment:
in future, please use https://godoc.org/github.com/juju/testing#Stub. it was basically made for exactly this. as a bonus, you can use it to check the arguments and also set errors to return from these mocks